clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -11133,6 +11134,25 @@ static void getNonTripleBasedSYCLPostLinkOpts(const ToolChain &TC,
 
   if (allowDeviceImageDependencies(TCArgs))
     addArgs(PostLinkArgs, TCArgs, {"-allow-device-image-dependencies"});
+
+  // For bfloat16 conversions LLVM IR devicelib, we only need to embed it
+  // when non-AOT compilation is used.
```

Comment:
Just for my understanding, since the comment reads `when non-AOT compilation is used` , do we need to check if we are not compiling for NVidia and AMD GPUS as well? `(!TC.getTriple().isNVPTX() || !TC.getTriple().isAMDGCN())`

Reply:
Hi, @srividya-sundaram 
We only use bfloat16 devicelib for Intel GPU/CPU device JIT compilation. The check is:
"if (TC.getTriple().isSPIROrSPIRV() && !TC.getTriple().isSPIRAOT())"

For NV and AMD device, TC.getTriple().isSPIROrSPIRV() will return false, the logic will be skipped.
Thanks very much.